### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by leon1995**
Add CI to run unittests and do release steps on tag push

### CODE REVIEW

**Review by ldeichmann — APPROVED** [review]

File: tests/test_client_device.py
Comment (ldeichmann) [inline tests/test_client_device.py:1]: Huh?

**Review by ldeichmann — CHANGES_REQUESTED** [review]

File: tests/test_client_device.py
Comment (leon1995) [reply tests/test_client_device.py:1]: thats also a way to get all tests passed. `git revert` does trick

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    runs-on: ubuntu-latest
+    needs: build
+    outputs:
+      DO_GITHUB_RELEASE: ${{ steps.detect-release.outputs.DO_GITHUB_RELEASE }}
+
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
```
Comment (a-kleinf) [inline .github/workflows/publish.yml:25]:
```suggestion
          name: distributions
```

Comment (leon1995) [reply .github/workflows/publish.yml]: done

**Review by ldeichmann — APPROVED** [review]

**Review by deichmab-draeger — APPROVED** [review]

**Review by a-kleinf — APPROVED** [review]

**Review by ldeichmann — APPROVED** [review]

**Review by deichmab-draeger — APPROVED** [review]

**Pull request merged**